src/tiled/mapdocument.cpp

```diff
@@ -246,6 +246,19 @@ void MapDocument::setCurrentLayer(Layer *layer)
             setCurrentObject(layer);
 }
 
+void MapDocument::setCurrentLayerByName(const QString *name)
+{
+    for (int i = 0; i < mLayerModel->rowCount(); ++i) {
```

Comment:
Please don't implement this based on the layer model. Also I think we don't need an extra method for this. We can use `Map::findLayer` to find the layer by name and then simply call `MapDocument::setCurrentLayer`.